controllers/apps/apicast_controller_status.go

```diff
@@ -95,6 +97,28 @@ func (r *APIcastReconciler) deploymentImage(ctx context.Context, cr *appsv1alpha
 	return deployment.Spec.Template.Spec.Containers[0].Image, nil
 }
 
+func (r *APIcastReconciler) reconcileHpaWarningMessage(conditions *[]metav1.Condition, cr *appsv1alpha1.APIcast) {
+	cond := &metav1.Condition{
+		Type:    appsv1alpha1.WarningConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  "HPA",
+		Message: "HorizontalPodAutoscaling (Hpa) enabled overrides values applied to request, limits and replicas",
+	}
+
+	// check if condition is already present
+	foundCondition := meta.FindStatusCondition(*conditions, "Warning")
+
+	// If hpa is enabled but the condition is not found, add it
+	if cr.Spec.Hpa && foundCondition == nil {
+		meta.SetStatusCondition(conditions, *cond)
+	}
```

Comment:
hmm thinking can I work with this? it is fine for values in the cr.Spec, but say you are trying to satify some other logical condition not tied to the CR like say a value in a secret, Are we going to have to bring those values into the status reconciler to add a message :thinking:

Reply:
I think HPA is unique since it's either on/off on the cr itself. So it's fine like that imo, but for more advanced warning we would need to make more changes